@navikt/core/react/src/form/virtualfocus/VirtualFocus.tsx

```diff
@@ -0,0 +1,251 @@
+   * The function that is run when the element gets
+   * virtual focus set to it.
+   */
+  onActive: () => void;
+  children: React.ReactNode;
+  /**
+   * set this to `-1` if you have an input element inside
+   * the Anchor that you would rather tab to directly instead
+   * of having to tab to the Anchor itself first.
+   *
+   * @default 0
+   */
+  tabIndex?: number;
+}
+
+export const VirtualFocusAnchor = forwardRef<
+  HTMLDivElement,
+  VirtualFocusAnchorProps
+>(({ children, pick, onActive, ...rest }, ref) => {
+  const { virtualFocusIdx, setVirtualFocusIdx, loop, uniqueId, containerRole } =
+    useVirtualFocusInternalContext();
+
+  const { register, descendants, index } = useVirtualFocusDescendant({
+    handlePick: () => {
+      pick();
+    },
+    handleOnActive: () => {
+      setVirtualFocusIdx(0);
+      onActive();
+    },
+  });
+
+  const mergedRefs = useMergeRefs(ref, register);
+
+  return (
+    <div
+      id={`virtualfocus-${uniqueId}-${index}`}
+      className="navds-virtualfocus-anchor"
+      role={containerRole}
+      tabIndex={-1}
+      aria-owns={`virtualfocus-${uniqueId}-content`}
+      aria-controls={`virtualfocus-${uniqueId}-content`}
+      aria-activedescendant={`virtualfocus-${uniqueId}-${virtualFocusIdx}`}
+      ref={mergedRefs}
+      onKeyDown={(event) => {
+        if (event.key === "ArrowDown") {
+          event.preventDefault();
+          const to_focus_descendant = descendants.next(virtualFocusIdx, loop);
+          const to_focus = to_focus_descendant?.node;
+
+          if (to_focus) {
+            to_focus_descendant.handleOnActive();
+          }
+        } else if (event.key === "ArrowUp") {
+          event.preventDefault();
+          const to_focus_descendant = descendants.prev(virtualFocusIdx, loop);
+          const to_focus = to_focus_descendant?.node;
+
+          if (to_focus) {
+            to_focus_descendant.handleOnActive();
+          }
+        } else if (event.key === "Enter") {
+          const curr = descendants.item(0);
+          if (curr?.handlePick) {
+            curr.handlePick();
+          }
+        }
+      }}
+      {...rest}
+    >
+      {children}
+    </div>
+  );
+});
+
+export const VirtualFocusContent = ({
+  children,
+}: {
+  children: React.ReactNode;
+}) => {
+  const { uniqueId } = useVirtualFocusInternalContext();
+  return (
+    <div
+      id={`virtualfocus-${uniqueId}-content`}
+      className="navds-virtualfocus-content"
+    >
+      {children}
+    </div>
+  );
+};
+
+export interface VirtualFocusItemProps
+  extends React.HTMLAttributes<HTMLDivElement> {
+  /**
+   * The function that is run when the element is focused
+   * (virtually, not actual focus, eg. set a border around an item)
+   */
+  onActive: () => void;
+  /**
+   * The function that is run when the focused element
+   * is to be picked (eg. do an actual search, change route... etc)
+   */
+  pick: () => void;
+  children: React.ReactNode;
+  itemRole?: React.AriaRole;
```

Comment:
Since `role` is part of `React.HTMLAttributes<HTMLDivElement>`, we can skip re-assigning it and just use `role`

Reply:
for containerRole, we should limit it to the subset of roles that use `aria-activedescendant` 🤔 

``` tsx
containerRole:
    | "combobox"
    | "grid"
    | "listbox"
    | "menu"
    | "menubar"
    | "radiogroup"
    | "tree"
    | "treegrid"
    | "tablist";
```

Reply:
We can do something like this: 
```
  role:
    | "combobox"
    | "grid"
    | "listbox"
    | "menu"
    | "menubar"
    | "radiogroup"
    | "tree"
    | "treegrid"
    | "tablist";
```
since the union is a subset of `role`, this will work while limiting the available options 🤞